client/src/components/CostChart.tsx

```diff
@@ -0,0 +1,496 @@
+                          />
+                          <circle
+                            key={`point-${res.name}-${i}`}
+                            cx={x}
+                            cy={y}
+                            r={5}
+                            fill={colors[idx]}
+                            className="hover:r-6 cursor-pointer transition-all duration-200"
+                            pointerEvents="none"
+                          />
+                        </g>
+                      );
+                    })}
+                  </g>
+                );
+              })}
+
+              {timeSeriesData.map((entry, i) => {
+                const x = (i / (timeSeriesData.length - 1)) * innerWidth;
+                const y = margin.top + innerHeight + 15;
+                return (
+                  <text
+                    key={entry.date}
+                    x={x}
+                    y={y}
+                    textAnchor="middle"
+                    transform={`rotate(-45, ${x}, ${y})`}
+                    className="text-xs text-gray-500"
+                    style={{ maxWidth: "40px", overflow: "hidden", textOverflow: "ellipsis" }}
+                  >
+                    {formatDate(entry.date)}
+                  </text>
+                );
+              })}
+            </svg>
+          </div>
+        </div>
+
+        {hoveredPoint && (
+          <div
+            className="pointer-events-none absolute"
+            style={{
+              left: `${hoveredPoint.x < margin.left + 80 ? hoveredPoint.x + 20 : hoveredPoint.x > margin.left + innerWidth - 80 ? hoveredPoint.x - 160 : hoveredPoint.x - 70}px`,
+              top: `${hoveredPoint.y < margin.top + 80 ? hoveredPoint.y + 20 : hoveredPoint.y - 80}px`,
+              zIndex: 10,
+            }}
+          >
+            <div
+              className="rounded border border-gray-200 bg-white p-3 text-sm shadow-lg"
+              style={{ width: "160px" }}
+            >
+              <div className="font-bold text-gray-800">{hoveredPoint.resourceName}</div>
+              <div className="text-gray-600">Date: {formatDate(hoveredPoint.date)}</div>
+              <div className="font-semibold text-gray-800">
+                Cost: ${hoveredPoint.value.toFixed(2)}
+              </div>
+            </div>
+          </div>
+        )}
+      </div>
+
+      <div className="mt-4 flex flex-wrap gap-4">
+        {sortedResources.map((res, i) => (
+          <div key={res.name} className="flex items-center">
+            <div className="mr-2 h-4 w-4 rounded" style={{ backgroundColor: colors[i] }} />
+            <span>
+              {res.name}
+              {loadingResources.includes(res.name) && (
+                <span className="ml-2 inline-block animate-pulse text-gray-500">(Loading...)</span>
+              )}
+            </span>
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+
+  const exportToCSV = () => {
```

Comment:
I guess we should not really try to do it client side, but instead offer a content-type handling or additional query parameter on the server-side
WDYT?

Reply:
I think this is a good idea, we can move this logic to the server later. I wanted to do a vertical slice in this PR, to set up some initial vision